anvil/src/eth/backend/mem/mod.rs

```diff
@@ -957,16 +989,30 @@ impl Backend {
         request: EthTransactionRequest,
         fee_details: FeeDetails,
         block_env: BlockEnv,
-    ) -> Result<(Return, TransactOut, u64, State), BlockchainError>
+    ) -> Result<(InstructionResult, Option<Output>, u64, State), BlockchainError>
     where
         D: DatabaseRef<Error = DatabaseError>,
     {
         let mut inspector = Inspector::default();
         let mut evm = revm::EVM::new();
         evm.env = self.build_call_env(request, fee_details, block_env);
         evm.database(state);
-        let (ExecutionResult { exit_reason, out, gas_used, .. }, state) =
-            evm.inspect_ref(&mut inspector);
+        let result_and_state = match evm.inspect_ref(&mut inspector) {
+            Ok(result_and_state) => result_and_state,
+            Err(e) => return Err(BlockchainError::EvmError(InstructionResult::FatalExternalError)),
```

Comment:
Also using this error as a placeholder—probably also need another variant for this type of inspector error